src/CommunityToolkit/Xamarin.CommunityToolkit/Startup/ToolkitPlatform.android.cs

```diff
@@ -17,8 +17,16 @@ internal static Context Context
 			get
 			{
 				var page = Forms.Application.Current.MainPage;
-				var renderer = page.GetRenderer();
-				return renderer.View.Context ?? throw new NullReferenceException($"{nameof(Context)} cannot be null");
+				if (page != null)
+				{
+					var renderer = page.GetRenderer();
+					return renderer.View.Context ?? throw new NullReferenceException($"{nameof(Context)} cannot be null");
+				}
+
+				// If MainPage isn't set yet, this is the only way to get Context
+#pragma warning disable CS0618 // Type or member is obsolete
+				return Forms.Forms.Context ?? throw new NullReferenceException($"{nameof(Context)} cannot be null");
```

Comment:
Can you revert this code? We don't want to use the obsolete member. If we don't have a MainPage it's because something is wrong.

Reply:
If I revert this code, our sample app will crash (this code is needed for this PR to work properly). 

That thing is that this line: `MainPage = new BaseNavigationPage(new WelcomePage());`
is not an atomic operations. First page is created, and only then it is set to `MainPage`. Issue is that if `NavigationBar` or `StatusBar` is applied to the `WelcomePage` (or to all pages through styles, like in the sample app), `MainPage` isn't set **yet** when they are evaluated.

Reply:
We'll need to find another solution then since we want to avoid using Obsolete members.

Reply:
I didn't find any. Except this one, or adding `Init()` method to XCT

Reply:
I need to dig into this, but we can listen for the "Renderer" property in Page control and do the action when this property changes

Reply:
The thing is that we don't have the page, since we can do `StatusBar.Color = Color.Black;`. So the only page `StatusBar` can get reliable access to is the `MainPage`. 

We can try to fetch page from element to which bindable property is attached to, but as you can see, user can change the style without using bindable property an all.

Reply:
Good thing is that this solution is here only for half a year. After XF merges with Xamarin.Essentials, we would be able to use `CurrentActivity` from there.

Or we can add initialization to XCT. XF and Xamarin.Essentials have it, why cannot we? 😄

Reply:
```ToolkitPlatform.Context``` ?

Reply:
@AndreiMisiukevich What do you mean? The issue is that 
- `ToolkitPlatform.Context` is null when `MainPage` is not set yet
- `MainPage` is created and only then it's set
- Methods in this PR can be used during `MainPage` initialization (like in its constructor or with style applied to all pages), and they use `ToolkitPlatform.Context` to get context

I updated `ToolkitPlatform.Context` to use obsolete `Forms.Forms.Context` in case if `MainPage` is null, and @pictos doesn't like that. I'm also not a fan of this approach, but it's the only way to get `Context` in this case that I found.

Reply:
We should wait for the `MainPage` to be instantiate to perform those changes.

Reply:
Discussed offline. Converted them into effects